### PR TITLE
[FIX] Fix exist? when caching nils

### PR DESCRIPTION
**Pull request by oded-zahavi**

### CODE REVIEW

**Comment by petergoldstein** [comment]
@oded-zahavi Is there a relevant test case?  If so, can you please add such a test?

**Comment by oded-zahavi** [comment]
`[11] pry(main)> Rails.cache.read('oded-zahavi', cache_nils: true)
=> #<Dalli::Server::NilObject:0x0000000278dcf8>
[12] pry(main)> Rails.cache.write('oded-zahavi', nil, cache_nils: true)
=> 15493227143084638208
[13] pry(main)> Rails.cache.read('oded-zahavi', cache_nils: true)
=> nil
[14] pry(main)> Rails.cache.exist?('oded-zahavi', cache_nils: true)
=> false
[15] pry(main)> Rails.cache.exist?('empty-eky', cache_nils: true)
=> true
[16] pry(main)> Rails.cache.exist?('empty-eky')
=> false
[17] pry(main)> Rails.cache.exist?('oded-zahavi')
=> false`

**Comment by oded-zahavi** [comment]
Hey. Thanks so much for addressing this PR. Of course. Since I'm caching nils, I read and write with `cache_nils: true`. Thus receiving `NilObject` before writing to cache (key does not exists) and `nil` after (I want to cache nils). However, when I want to just ask if key exists I'm getting a wrong indication (lines 4+5). 
`
[1] pry(main)> Rails.cache.read('oded-zahavi', cache_nils: true)
=> #<Dalli::Server::NilObject:0x0000000278dcf8>
[2] pry(main)> Rails.cache.write('oded-zahavi', nil, cache_nils: true)
=> 15493227143084638208
[3] pry(main)> Rails.cache.read('oded-zahavi', cache_nils: true)
=> nil
[4] pry(main)> Rails.cache.exist?('oded-zahavi', cache_nils: true)
=> false
[5] pry(main)> Rails.cache.exist?('empty-key', cache_nils: true)
=> true
[6] pry(main)> Rails.cache.exist?('empty-key')
=> false
[7] pry(main)> Rails.cache.exist?('oded-zahavi')
=> false
`

**Comment by petergoldstein** [comment]
@oded-zahavi I don't really understand your previous comment, and it doesn't address my question.  There is a test suite in the Dalli repository.  Please add a test that covers this case.  I won't be able to consider the PR without one.

Thanks.

**Comment by oded-zahavi** [comment]
Thanks @petergoldstein I have updated my pull request and added tests to cover the fix.

File: test/test_active_support.rb [hunk 1]
```diff
@@ -97,6 +97,38 @@ def self.it_with_and_without_local_cache(message, &block)
...
+          dvalue = @dalli.exist?('nil')
+          assert_equal true, dvalue
+        end
+      end
+
+      it 'return false when caching non-nil and cache_nils: false' do
+        with_cache cache_nils: false do
+          @dalli.write('nil', 'nil')
+          dvalue = @dalli.exist?('nil')
+          assert_equal true, dvalue
```
Comment (petergoldstein) [inline test/test_active_support.rb]: This value seems to conflict with the description of the test.

**Comment by petergoldstein** [comment]
@oded-zahavi Can you please rebase and address the open question?  Then I can consider merging.  Thanks.

**Pull request closed without merging**